src/main/scala/cromwell/engine/backend/runtimeattributes/CromwellRuntimeAttributes.scala

```diff
@@ -24,156 +27,159 @@ case class CromwellRuntimeAttributes(docker: Option[String],
 object CromwellRuntimeAttributes {
   private val log = LoggerFactory.getLogger("RuntimeAttributes")
 
-  def apply(wdlRuntimeAttributes: RuntimeAttributes, backendType: BackendType): CromwellRuntimeAttributes = {
-    val attributeMap = AttributeMap(wdlRuntimeAttributes.attrs)
+  def apply(wdlRuntimeAttributes: RuntimeAttributes, backendCall: BackendCall): Try[CromwellRuntimeAttributes] = {
+    for {
+      keys <- validateKeys(wdlRuntimeAttributes, backendCall.backend.backendType)
+      attributes <- TryUtil.sequenceMap(wdlRuntimeAttributes.evaluate(backendCall.lookupFunction(backendCall.locallyQualifiedInputs), backendCall.engineFunctions))
+      validatedAttributes <- validateRuntimeAttributes(attributes)
+    } yield validatedAttributes
+  }
+
+  def validateKeys(wdlRuntimeAttributes: RuntimeAttributes, backendType: BackendType): Try[RuntimeAttributes] = {
```

Comment:
How would you structure the `apply()` method if this returned an `ErrorOr`?  I could either:

* Change the return type of `RuntimeAttributes.evaluate` to return an `ErrorOr` (but then wdl4s returns scalaz objects from its public API).  This also conflicts with your other suggestion of throwing an exception from that method.
* Convert the return value of `RuntimeAttributes.evaluate` to an `ErrorOr` in `apply`

Any other suggestions?